### DISCUSSION TITLE
Step 7 script `Build ipa for distribution` exited with status code 9

### DISCUSSION THREAD

**Discussion by taivo202** [codemagic-ci-cd/codemagic-docs#2790] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Start new build

### Expected results

Publish success

### Actual results

I got build failed :(((. help me plzzzzz
![image](https://github.com/user-attachments/assets/c815e14b-f5ce-44b2-ba94-5a5862ec79a0)

**Comment by taivo202** [comment]
buildId: 6711dcc85acd5a94f59045f4

**Reply by dam-ease** [reply]
Hello @taivo202,
In your Build ipa for distribution step, I see your command is:
```
cd ios
xcode-project build-ipa -v \
  --workspace "$XCODE_WORKSPACE" \
  --scheme "$XCODE_SCHEME" \
  --archive-flags="-destination 'generic/platform=iOS'" \
  --archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"
```
Can you please remove `--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"`? 
It is not required and re-run the build.
Also, you don't have to run `cd ios` command at every step, this can be removed as well.

**Reply by taivo202** [reply]
Tks for reply, i tried remove --archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO" but it still doesnt working. build fail ipa with status 65

**Reply by dam-ease** [reply]
Hey @taivo202,
I looked through your recent builds and I still see the argument, please can you share the buildID of the build where this argument was removed?

**Reply by taivo202** [reply]
Hey @dam-ease, buildId: 6711c118c8a35646998a3938
Can you help me plzz huhu :(((

**Reply by taivo202** [reply]
I tried everything from dicussions but it still doesnt work

**Reply by dam-ease** [reply]
Hello @taivo202.
I can't seem to reproduce this on my end. Are you able to build your project locally? If yes, please can you share the version of Xcode you are making use of?
Kindly try building this step again with below command.
```
xcode-project build-ipa -v \
  --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
  --scheme "$XCODE_SCHEME" \
  --archive-flags="-destination 'generic/platform=iOS'" 
```
Please also try removing the slash and let's see if that helps. I added the -v argument so we can get more detailed logs.

**Reply by dam-ease** [reply]
Hello @taivo202,
Any update for me on this?
If there is anything else I can assist you with, please don't hesitate to let me know.

**Reply by taivo202** [reply]
Hello @dam-ease , i tried follow your code but still build failed "Step 7 script `Build ipa for distribution` exited with status code 65".  Please help me. Tks so much
*buildId: 67286a19f2db5605b104ea50

**Reply by taivo202** [reply]
I use Xcode 16.0 to build success your project locally.

**Reply by taivo202** [reply]
tks @dam-ease , i found reason and fixed it. Tks so much. Love u

**Reply by ariefrhmn** [reply]
hello @taivo202, may i know what you doin to fix the problem? i think ive got same issue here https://github.com/orgs/codemagic-ci-cd/discussions/2801
your help is greatly appreciated <3
thanks
